weave/integrations/bedrock/bedrock_sdk.py

```diff
@@ -147,6 +185,108 @@ def postprocess_output_invoke(
     return outputs
 
 
+def postprocess_output_invoke_agent(
+    outputs: Optional[dict[str, Any]],
+) -> Optional[dict[str, Any]]:
+    """Process the outputs for invoke_agent, extracting essential info from EventStream while preserving it for user."""
+    if outputs is None:
+        return None
+
+    if "completion" in outputs:
+        # Create a copy of outputs for logging
+        outputs_copy = {
+            k: copy.deepcopy(v) for k, v in outputs.items() if k != "completion"
+        }
+
+        # Process the EventStream to extract essential content for logging
+        all_events = []
+        extracted_text = ""
+        foundation_model = None
+        usage_info = None
+
+        try:
+            # Consume the completion EventStream to collect events
+            completion = outputs["completion"]
+            if hasattr(completion, "__iter__"):
```

Comment:
Do you have to do this?  Can't you check the type?

Reply:
we can actually just do it,